SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -2987,28 +2512,59 @@ void CEulerSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container
 
   /*--- Update the angle of attack at the far-field for fixed CL calculations (only direct problem). ---*/
 
-  if ((fixed_cl) && (!disc_adjoint) && (!cont_adjoint)) { SetFarfield_AoA(geometry, solver_container, config, iMesh, Output); }
+  if (fixed_cl && !disc_adjoint && !cont_adjoint) {
+    SU2_OMP_MASTER
+    SetFarfield_AoA(geometry, solver_container, config, iMesh, Output);
+    SU2_OMP_BARRIER
```

Comment:
Making those methods protected allows the decision to call them in parallel or serial to be centralized all in one function (Preprocessing in this case).
Similar, but less extensive, changes were made to the turbulence solvers.